docs/design/resolve-divergence.md

```diff
@@ -0,0 +1,104 @@
+# Resolve-Divergence Command Design
+
+Authors: [David Rieber](mailto:drieber@google.com)
+
+**Summary:** This document is a high-level proposal for a new `jj resolve-divergence` command to help users resolve (or reduce) divergence.
+
+## Objective
+
+Divergence is confusing to our users. In many common scenarios we believe it should be possible to completely eliminate divergence with the help of this command. However not all theoretically possible cases of divergence are easy to solve. It is not a goal to solve every single case, but instead the most common ones.
+
+Why a new command? why not automatically resolve divergence during every existing command? Some backends (e.g. the Google backend) use a cloud architecture. In such backends multiple clocks are involved. Because of this it is not always possible to establish a total chronological ordering of jj operations. Because of this we believe it is best to introduce a new command. The command allows us to obtain input from the user to guide the algorithm when we lack concrete rules or heuristics to make certain decisions.
+
+The command may be unable to make any changes, or may only make partial progress. It should produce informative messages to summarize any changes made, and/or guide the user towards possible next steps. It is a soft goal to make the algorithm deterministic, mainly to make testing easier.
```

Comment:
nit: I don't think a non-deterministic algorithm is viable from both a library and CLI standpoint.